tests/test_optics.py

```diff
@@ -53,3 +54,77 @@ def test_sunrise_sunset_fail():
+
+
+@pytest.mark.parametrize("percentile", [5, 50, 95])
+def test_backscatter_dark_count(percentile):
+    from glidertools.optics import backscatter_dark_count
+
+    # create some synthetic data
+    bbp = np.array([0.002, 0.0006, 0.0005, 0.0005, 0.0005])
+    depth = np.array([50, 150, 210, 310, 350])
+    # select only depths between 200 and 400
+    mask = (depth > 200) & (depth < 400)
+    # expected output
+    expected_bbp_dark = bbp - np.nanpercentile(bbp[mask], percentile)
+    bbp_dark = backscatter_dark_count(bbp, depth, percentile)
+    np.testing.assert_allclose(expected_bbp_dark, bbp_dark)
+
+
+@pytest.mark.parametrize("percentile", [5, 50, 95])
+def test_backscatter_dark_count_negative(percentile):
+    from glidertools.optics import backscatter_dark_count
+
+    # create some synthetic data
+    bbp = np.array(
+        [0.002, 0.0006, 0.005, 0.005, 0.0004]
+    )  # this will result in negative values that should be zeroed out
+    depth = np.array([50, 150, 210, 310, 350])
+    bbp_dark = backscatter_dark_count(bbp, depth, percentile)
+    # in this case we just want to check if none of the values is negative!
+    assert np.all(bbp_dark >= 0)
+
+
+@pytest.mark.parametrize("percentile", [5, 50, 95])
+def test_flr_dark_count(percentile):
+    from glidertools.optics import fluorescence_dark_count
+
+    # create some synthetic data
+    flr = np.array([200.0, 100.0, 52.0, 52.0])
+    depth = np.array([20, 50, 310, 350])
+    # select only depths between 200 and 400
+    mask = (depth > 300) & (depth < 400)
+    # expected output
+    expected_flr_dark = flr - np.nanpercentile(flr[mask], percentile)
+    flr_dark = fluorescence_dark_count(flr, depth, percentile)
+    np.testing.assert_allclose(expected_flr_dark, flr_dark)
+
+
+@pytest.mark.parametrize("percentile", [5, 50, 95])
+def test_flr_dark_count_negative(percentile):
+    from glidertools.optics import fluorescence_dark_count
+
+    # create some synthetic data
+    flr = np.array([200.0, 100.0, 152.0, 151.0])
+    # this will result in negative values that should be zeroed out
+    depth = np.array([20, 50, 310, 350])
+    flr_dark = fluorescence_dark_count(flr, depth, percentile)
+    # in this case we just want to check if none of the values is negative!
+    assert np.all(flr_dark >= 0)
+
+
+@pytest.mark.parametrize("percentile", [90])
+def test_par_dark_count(percentile):
+    from glidertools.optics import par_dark_count
+    from pandas import date_range
+
+    # create some synthetic data
+    par = np.array([34, 23.0, 0.89, 0.89])
+    depth = np.array([10, 20, 310, 350])
+    time = date_range("2018-12-01 10:00", "2018-12-03 00:00", 4)
+    # expected output
+    expected_par_dark = par - np.nanmedian(
+        np.nanpercentile(par[-1], percentile)
+    )  # only use values in the 90% percentile of depths and between 23:00 and 01:00
+    par_dark = par_dark_count(par, depth, time, percentile)
+    np.testing.assert_allclose(expected_par_dark, par_dark)
```

Comment:
Same as my comment above but for `par_dark_count`

Reply:
Just noting this here. It took me a while to get this to work - the trick was changing the current version of warnings <raise UserWarning ""> to  <warnings.warn(, UserWarning)>. I think this will need fixing throughout the older version of GliderTools where warnings are implemented for this type of unit test to work.

Reply:
😬 yeah. I have never seen warnings being implemented like that. I think this should be tackled in a separate PR though. Would you mind raising an issue about this? We can discuss how to fix this over there.